cluster-controller/src/main/java/io/strimzi/controller/cluster/resources/TopicController.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright 2017-2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.controller.cluster.resources;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.api.model.EnvVar;
+import io.fabric8.kubernetes.api.model.EnvVarBuilder;
+import io.fabric8.kubernetes.api.model.extensions.Deployment;
+import io.vertx.core.json.JsonObject;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Represents the topic controller deployment
+ */
+public class TopicController extends AbstractCluster {
+
+    private static final String NAME_SUFFIX = "-topic-controller";
+
+    // Topic Controller configuration
+    protected TopicControllerConfig config;
+
+    // Port configuration
+    private static final int HEALTHCHECK_PORT = 8080;
+    private static final String HEALTHCHECK_PORT_NAME = "hcheck-port";
+
+    // Configuration defaults
+    private static final String DEFAULT_IMAGE = "strimzi/topic-controller:latest";
+    private static final int DEFAULT_REPLICAS = 1;
```

Comment:
Do we need things like default replicas? It should have always 1 replica.

Reply:
The base `AbstractCluster` class use the `replicas` field in the `createDeployment` for setting the number of replicas so it should be initialized to a meaningful value (1 in our case).